toucan_connectors/redshift/redshift_connector.py

```diff
@@ -0,0 +1,97 @@
+from contextlib import suppress
+from typing import Dict
+
+import pandas as pd
+import redshift_connector
+from pydantic import Field, SecretStr, create_model
+
+from toucan_connectors.common import ConnectorStatus
+from toucan_connectors.toucan_connector import (
+    strlist_to_enum,
+    ToucanConnector,
+    ToucanDataSource,
+)
+
+
+class RedshiftDataSource(ToucanDataSource):
+    database: str = Field(..., description='The name of the database you want to query')
+
+    table: str = Field(None, description='The name of the data table that you want to ')
+
+    query: Dict = Field(None, description='An object describing a simple select query')
+
+    @classmethod
+    def get_form(cls, connector: 'RedshiftConnector', current_config):
+        constraints = {}
+        query = """select oid as database_id, datname as database_name, datallowconn as allow_connect from pg_database order by oid;"""
+        with suppress(Exception):
+            connection = redshift_connector.connect(
+                **connector.get_connection_params(
+                    database=current_config.get('database', 'redshift')
+                )
+            )
+            with connection.cursor() as cursor:
+                cursor.execute(query)
+                res = cursor.fetchall()
+                available_dbs = [db_name for (db_name,) in res]
+                constraints['database'] = strlist_to_enum('database', available_dbs)
+
+        return create_model('FormSchema', **constraints, __base__=cls).schema()
+
+
+class RedshiftConnector(ToucanConnector):
+    data_source_model: RedshiftDataSource
+
+    dbname: str = Field(..., description='The database name.')
+    user: str = Field(..., description='Your login username.')
+    password: SecretStr = Field(None, description='Your login password')
+    host: str = Field(None, description='IP address or hostname.')
+    port: int = Field(..., description='port value of 5439 is specified by default')
```

Comment:
Je ne vois pas où est-ce que le port par défaut est définit. A mon avis il vaut mieux ne pas mettre de port par défaut.  

```suggestion
    port: int = Field(..., description='The listening port of your Redshift Database')
```